astropy/io/fits/header.py

```diff
@@ -350,13 +350,29 @@ def fromstring(cls, data, sep=''):
 
         cards = []
 
+        if not isinstance(sep, str):
+            sep = sep.decode('latin1')
+
         # If the card separator contains characters that may validly appear in
         # a card, the only way to unambiguously distinguish between cards is to
         # require that they be Card.length long.  However, if the separator
         # contains non-valid characters (namely \n) the cards may be split
         # immediately at the separator
         require_full_cardlength = set(sep).issubset(VALID_HEADER_CHARS)
 
+        if isinstance(data, bytes):
+            CONTINUE = b'CONTINUE'
+            END = b'END'
+            end_card = END_CARD.encode('ascii')
+            if not isinstance(sep, bytes):
+                sep = sep.encode('latin1')
+            join = lambda i: b''.join(i)
```

Comment:
Just a suggestion: Instead of using a `lambda` you could define the empty string / bytes the same way you have done with END and CONTINUE:  eg. `empty = b''` and `empty = ''` then `join = empty.join` after the conditional.  Seems like it may be a bit neater.

Reply:
Yes, I agree.

Reply:
Done.